cmake/OpenCVConfig.cmake

```diff
@@ -47,7 +47,7 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
```

Comment:
I'm actually thinking of the the Windows megapack. The problem there is to not just find a package, but to find the _correct_ package. With your change, projects that don't define `BUILD_SHARED_LIBS` or `OpenCV_STATIC` would suddenly switch from the dynamic build to the static one. I'd rather avoid that.